app.js

```diff
@@ -287,12 +317,26 @@ router.post('/account/sendRecoveryMessage', async ctx => {
     }
 
     const account = await models.Account.findOne({ where: { accountId } });
-    const [recoveryMethod] = await account.getRecoveryMethods({ where: {
-        kind: method.kind,
-        detail: method.detail
-    }});
 
-    await recoveryMethod.update({ publicKey, securityCode: null });
+    if (isNew) {
+        // remove all recovery methods
+        const allRecoveryMethods = await account.getRecoveryMethods();
+        for (const rm of allRecoveryMethods) {
+            await rm.destroy();
+        }
+        // set up the official recovery method for this account
+        await account.createRecoveryMethod({
```

Comment:
isn't `RecoveryMethod` instance already created in `/account/initializeRecoveryMethodForTempAccount`?

Reply:
IMHO it was easier to destroy all recovery methods and create the one we will send link to vs. try to select all recovery methods that are NOT the one we send recovery link to.

Reply:
won't it be just `if (rm.public != publicKey) {` in a loop?

Reply:
you're right! cleaner now!